src/platform/datapath_raw_socket.c

```diff
@@ -492,15 +499,15 @@ CxPlatFramingTransportChecksum(
 _IRQL_requires_max_(DISPATCH_LEVEL)
 void
 CxPlatDpRawSocketAckFin(
-    _In_ CXPLAT_SOCKET* Socket,
+    _In_ CXPLAT_SOCKET_RAW* Socket,
     _In_ CXPLAT_RECV_DATA* Packet
     )
 {
     CXPLAT_DBG_ASSERT(Socket->UseTcp);
 
     CXPLAT_ROUTE* Route = Packet->Route;
     CXPLAT_SEND_CONFIG SendConfig = { Route, 0, CXPLAT_ECN_NON_ECT, 0 };
-    CXPLAT_SEND_DATA *SendData = CxPlatSendDataAlloc(Socket, &SendConfig);
+    CXPLAT_SEND_DATA *SendData = (CXPLAT_SEND_DATA*)CxPlatSendDataAlloc((CXPLAT_SOCKET*)Socket, &SendConfig);
```

Comment:
What does `CxPlatSendDataAlloc` return? Usually C doesn't make you cast things if it's a `void*`.

Reply:
wow, this looks like simply bug....